In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
#from codice.EDA_data.open_file import artwork_cards_df

# OPEN PROVVISSORIO

In [2]:
#aggiunto per permettermi di usare %store fino a quando non capisco dove mi salva le cose (%store)
# scryfall-artwork-cards

artwork_cards_df = pd.read_json("scryfall-artwork-cards.json")

In [4]:
artwork_cards_df_bu = artwork_cards_df

In [3]:
artwork_cards_df.shape

(25292, 71)

# VALUE NULL

In [6]:
artwork_cards_df.head(5)

object                                    id  \
0   card  dbcdbf7a-9294-47ad-9f93-c16b78c7463a   
1   card  acb3ce9b-ee4f-410a-8db3-e87aeb0a4444   
2   card  17b2ed72-d0f0-4d8d-bb5e-dce08d157466   
3   card  0faa9eea-fbf1-41f7-9def-1ec3d5134a53   
4   card  ecbeac44-9392-4522-8ff5-87079386bd0a   

                              oracle_id multiverse_ids  tcgplayer_id  \
0  cd6250ae-9079-4a62-8a70-0d94fbac21bc             []      200607.0   
1  ab0dfae5-b9d4-417b-8a0d-2525ae3a73b9             []      200606.0   
2  7264d3b3-fd46-4ea3-a85d-f0b068c331ad             []      200605.0   
3  e6284fb3-7cf8-4730-b156-10085b70b0e8             []      200604.0   
4  43296f8b-58d9-446e-a538-1c4921552c41             []      200603.0   

                      name lang released_at  \
0        Earthshaker Giant   en  2019-11-15   
1             Fiendish Duo   en  2019-11-15   
2         Calculating Lich   en  2019-11-15   
3  Sphinx of Enlightenment   en  2019-11-15   
4        Highcliff Felidar   en  2019-11-15   

                                                 uri  \
0  https://api.scryfall.com/cards/dbcdbf7a-9294-4...   
1  https://api.scryfall.com/cards/acb3ce9b-ee4f-4...   
2  https://api.scryfall.com/cards/17b2ed72-d0f0-4...   
3  https://api.scryfall.com/cards/0faa9eea-fbf1-4...   
4  https://api.scryfall.com/cards/ecbeac44-9392-4...   

                                        scryfall_uri  ... loyalty  mtgo_id  \
0  https://scryfall.com/card/gn2/5/earthshaker-gi...  ...     NaN      NaN   
1  https://scryfall.com/card/gn2/4/fiendish-duo?u...  ...     NaN      NaN   
2  https://scryfall.com/card/gn2/3/calculating-li...  ...     NaN      NaN   
3  https://scryfall.com/card/gn2/2/sphinx-of-enli...  ...     NaN      NaN   
4  https://scryfall.com/card/gn2/1/highcliff-feli...  ...     NaN      NaN   

  color_indicator printed_name  printed_type_line printed_text mtgo_foil_id  \
0             NaN          NaN                NaN          NaN          NaN   
1             NaN          NaN                NaN          NaN          NaN   
2             NaN          NaN                NaN          NaN          NaN   
3             NaN          NaN                NaN          NaN          NaN   
4             NaN          NaN                NaN          NaN          NaN   

  life_modifier hand_modifier variation_of  
0           NaN           NaN          NaN  
1           NaN           NaN          NaN  
2           NaN           NaN          NaN  
3           NaN           NaN          NaN  
4           NaN           NaN          NaN  

[5 rows x 71 columns]

In [7]:
artwork_cards_df.isna().sum()[50:70]

booster                  0
story_spotlight          0
related_uris             0
edhrec_rank           2582
watermark            22556
frame_effects        24530
card_faces           24920
preview              24171
arena_id             22012
all_parts            23597
promo_types          23644
loyalty              24976
mtgo_id               5771
color_indicator      25261
printed_name         25232
printed_type_line    25250
printed_text         25250
mtgo_foil_id          8480
life_modifier        25175
hand_modifier        25175
dtype: int64

# REMOVE / FILLNA

TCGPLAYER_ID NON MI INTERESSA ID DI UN STORE ONLINE

In [8]:
artwork_cards_df.drop(columns='tcgplayer_id', inplace=True)

oracle_text -> card text; fillna with no oracle text

In [9]:
artwork_cards_df.oracle_text.fillna('No oracle text',inplace=True)

power             12958 -> card sorcery,instant,enchantment and etc.

toughness         12958 -> card sorcery,instant,enchantment and etc.

i dont't update the Nan

flavor_text -> card lore 

fillna -> No flavor text

In [10]:
artwork_cards_df.flavor_text.fillna('No flavor text', inplace=True)

illustration_id -> The unique identifier generated by Scryfall for the card artwork that remains consistent across reprints. Newly previewed cards may not have this property yet.

edhrec_rank -> rank for commander format

In [11]:
artwork_cards_df.edhrec_rank.fillna(str('not ranked'), inplace=True)

C:\Users\elped\AppData\Local\Temp\ipykernel_15452\243432457.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'not ranked' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  artwork_cards_df.edhrec_rank.fillna(str('not ranked'), inplace=True)


Watermark ->The name of the watermark on the card.

In [12]:
artwork_cards_df.drop(columns='watermark',inplace=True)

(25292, 71)

REMOVE

column -> Nan value

frame_effects        24530 -> design card (remove)

card_faces           24920 -> CAPIRE

preview              24171 -> dove hanno mostrato per la prima volta la carta (remove)

arena_id             22012 -> id per mtg arena (remove)

all_parts            23597 -> tutte le carte per una combo (CAPIRE) fillna -> no combo

promo_types          23644 -> carte promo (remove)

loyalty              24976 -> plainswalker stats (CAPIRE) importante per plain

mtgo_id               5771 -> id per magic online (remove)

color_indicator      25261 -> color indicatore usually for double face (remove)

printed_name         25232 -> (remove)

printed_type_line    25250 -> (remove)

printed_text         25250 -> (remove)

mtgo_foil_id          8480 -> id foil card magic online (remove)

life_modifier        25175 -> (remove) The starting life total modifier. A + or - character precedes a number. Used only on cards with "Vanguard" in its types.

hand_modifier        25175 -> (remove) The starting maximum hand size total modifier. A + or - character precedes a number. Used only on cards with "Vanguard" in its types.

variation_of         25282 -> (remove) A list of card uuid's of this card with alternate printings in the same set, excluding Un‑sets.


In [13]:
artwork_cards_df.drop(['frame_effects','preview','promo_types','mtgo_id','color_indicator','printed_name','printed_type_line','printed_text','mtgo_foil_id','life_modifier','hand_modifier','variation_of','arena_id'], axis=1, inplace = True)

In [14]:
artwork_cards_df.all_parts.fillna('no combo',inplace=True)

In [15]:
artwork_cards_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25292 entries, 0 to 25291
Data columns (total 56 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   object             25292 non-null  object        
 1   id                 25292 non-null  object        
 2   oracle_id          25292 non-null  object        
 3   multiverse_ids     25292 non-null  object        
 4   name               25292 non-null  object        
 5   lang               25292 non-null  object        
 6   released_at        25292 non-null  datetime64[ns]
 7   uri                25292 non-null  object        
 8   scryfall_uri       25292 non-null  object        
 9   layout             25292 non-null  object        
 10  highres_image      25292 non-null  bool          
 11  image_uris         25095 non-null  object        
 12  mana_cost          25095 non-null  object        
 13  cmc                25292 non-null  float64       
 14  type_l

In [16]:
artwork_cards_df.all_parts[artwork_cards_df.all_parts != 'no combo'].iloc[0]

[{'object': 'related_card',
  'id': '867e245b-6153-479c-9e09-682afb017ff3',
  'component': 'combo_piece',
  'name': 'Gluttonous Troll',
  'type_line': 'Creature — Troll',
  'uri': 'https://api.scryfall.com/cards/867e245b-6153-479c-9e09-682afb017ff3'},
 {'object': 'related_card',
  'id': '13e8f518-1c0d-4c01-a397-71481839d02b',
  'component': 'token',
  'name': 'Food',
  'type_line': 'Token Artifact — Food',
  'uri': 'https://api.scryfall.com/cards/13e8f518-1c0d-4c01-a397-71481839d02b'}]

In [17]:
artwork_cards_clean_df = artwork_cards_df

In [18]:
artwork_cards_clean_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25292 entries, 0 to 25291
Data columns (total 56 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   object             25292 non-null  object        
 1   id                 25292 non-null  object        
 2   oracle_id          25292 non-null  object        
 3   multiverse_ids     25292 non-null  object        
 4   name               25292 non-null  object        
 5   lang               25292 non-null  object        
 6   released_at        25292 non-null  datetime64[ns]
 7   uri                25292 non-null  object        
 8   scryfall_uri       25292 non-null  object        
 9   layout             25292 non-null  object        
 10  highres_image      25292 non-null  bool          
 11  image_uris         25095 non-null  object        
 12  mana_cost          25095 non-null  object        
 13  cmc                25292 non-null  float64       
 14  type_l

# SALVO DATA SET PULITO

In [19]:
%store artwork_cards_clean_df

Stored 'artwork_cards_clean_df' (DataFrame)
